src/Crypto/EncryptionKey.php

```diff
@@ -0,0 +1,39 @@
+<?php declare(strict_types = 1);
+
+namespace Cmnty\Push\Crypto;
+
+use InvalidArgumentException;
+
+class EncryptionKey implements KeyingMaterial
+{
+    /**
+     * @var BinaryString
+     */
+    private $binaryString;
+
+    /**
+     * Create encryption key.
+     *
+     * @param BinaryString $binaryString
+     *
+     * @throws InvalidArgumentException When the encryption key is not the correct length.
+     */
+    public function __construct(BinaryString $binaryString)
+    {
+        if ($binaryString->getLength() != 16) {
+            throw new InvalidArgumentException('EncryptionKey could not be created: incorrect length.');
+        }
+
+        $this->binaryString = $binaryString;
+    }
+
+    /**
+     * Get raw bytes.
+     *
+     * @return string
+     */
+    public function getRawKeyMaterial() : string
```

Comment:
They are similar. I started with the `KeyingMaterial` interface and the `getRawKeyMaterial` method for all the keying material. (e.g. the public key, shared secret, encyption key, or psuedo random key). Then i added the second interface with the method `getRawBytes`. I left them separated because that way you cant accidentally use a regular binary string as keying material and vice versa.